lib/IRGen/GenDistributed.cpp

```diff
@@ -269,9 +269,17 @@ static CanSILFunctionType getAccessorType(IRGenModule &IGM,
     auto *actor = getDistributedActorOf(Target);
     assert(actor);
 
-    for (auto *genericParam : actor->getInnermostGenericParamTypes())
+    for (auto *genericParam : actor->getInnermostGenericParamTypes()) {
       genericParams.push_back(genericParam);
 
+      // and also forward all requirements this generic parameter might have.
+      for (auto req : actor->getGenericRequirements()) {
+        if (req.getFirstType()->isEqual(genericParam)) {
+          genericRequirements.push_back(req);
```

Comment:
Please cherry pick the corrected version from main that uses buildGenericSignature() and preserves all generic requirements, not just those whose subject type is exactly equal to an innermost generic parameter.

Reply:
Thanks Slava, slight de-javu actually indeed... we worked on this in https://github.com/apple/swift/pull/69502
I'll re-evaluate what we're missing and if we even needed this type change still.